addons/mail/static/src/core/common/thread_model.js

```diff
@@ -587,6 +590,49 @@ export class Thread extends Record {
         return this.memberCount - this.channelMembers.length;
     }
 
+    isReadonly(fetchAccessRights = true) {
```

Comment:
I would rename to `canPost`, so it's easier to reason with positives. It's easier to see we can post if we have write access or if read access when authorised.